src/main/scala/viper/server/core/MessageReportingTask.scala

```diff
@@ -0,0 +1,33 @@
+package viper.server.core
+
+import viper.server.vsi.MessageStreamingTask
+import viper.silver.ast.Program
+import viper.silver.reporter.{EntityFailureMessage, EntitySuccessMessage, Message, Reporter}
+
+trait MessageReportingTask extends MessageStreamingTask[Program] with ViperPost {
+
+//  private val _reporter:
+
+  protected def enqueueMessage(msg: Message): Unit = {
+    super.enqueueMessage(pack(msg))
+  }
+
+  // Implementation of the Reporter interface used by the backend.
+  class ActorReporter(tag: String) extends Reporter {
```

Comment:
I agree that the name is not ideal. Technically it makes sense since an "actor reporter" is a "reporter", the word "actor" is meant as an adjective here. So, the logic is this: an ActorReporter is a reporter that uses actors in its implementation. 

Any suggestions for a better name?

Reply:
Okay, nope I don't have a better name in mind either